adafruit_lsm303_accel.py

```diff
@@ -394,3 +410,42 @@ def _read_bytes(device, address, count, buf):
         with device as i2c:
             buf[0] = address & 0xFF
             i2c.write_then_readinto(buf, buf, out_end=1, in_end=count)
+
+    @property
+    def fifo_mode(self):
+        """Sets the fifo mode of the sensor. The mode must be a `FifoMode`."""
+        return self._cached_fifo_mode
+
+    @fifo_mode.setter
+    def fifo_mode(self, value):
+        if value < 0 or value > 2:
+            raise AttributeError("fifo mode must be a 'FifoMode'")
+        if value == 4:
+            self._fifo_enable = 0
+        else:
```

Comment:
The logic in the if statements here seems to contradict each other. As far as I understand it, the intention is that of these are valid values:
```
FIFO_MODE_BYPASS = const(0)
FIFO_MODE_FIFO = const(1)
FIFO_MODE_STREAM = const(2)
FIFO_MODE_STREAM_TO_FIFO = const(3)
FIFO_MODE_DISABLE = const(4)
```
But the first if statement would cause both `FIFO_MODE_STREAM_TO_FIFO` and `FIFO_MODE_DISABLE` to raise the exception since they are greater than `2`.

In the following if statement it checks if the value is `4`, which it couldn't ever be, because the prior if statement would cause it to raise that exception.